src/hooks/useGetWalletNfts.ts

```diff
@@ -54,66 +52,84 @@ const useGetWalletNfts = () => {
   const { account } = useWeb3React()
   const { lastUpdated } = state
 
-  useEffect(() => {
-    const fetchNfts = async () => {
-      try {
-        const balanceOf = await pancakeRabbitsContract.methods.balanceOf(account).call()
-
-        if (balanceOf > 0) {
-          const getTokenIdAndBunnyId = async (index: number) => {
-            try {
-              const { tokenOfOwnerByIndex, getBunnyId, tokenURI } = pancakeRabbitsContract.methods
-              const tokenId = await tokenOfOwnerByIndex(account, index).call()
-              const [bunnyId, tokenUri] = await makeBatchRequest([getBunnyId(tokenId).call, tokenURI(tokenId).call])
-
-              return [Number(bunnyId), Number(tokenId), tokenUri]
-            } catch (error) {
-              return null
-            }
-          }
+  /**
+   * Helper function to get tokenIds by identifier
+   */
+  const getTokenIdsByIdentifier = (identifier: Nft['identifier']) => {
+    return state.tokenIds[identifier] || []
+  }
 
-          const tokenIdPromises = []
+  /**
+   * Helper function to return all nfts in the config that are in the tokenIds
+   */
+  const getNftsInWallet = () => {
+    const identifiers = Object.keys(state.tokenIds)
+    return Nfts.filter((nft) => identifiers.includes(nft.identifier))
+  }
 
-          for (let i = 0; i < balanceOf; i++) {
-            tokenIdPromises.push(getTokenIdAndBunnyId(i))
+  useEffect(() => {
+    const fetchCollectibles = async () => {
+      // For each nft source get nft data
+      Object.keys(nftSources).forEach(async (nftSourceType) => {
```

Comment:
The main structural change in this hook is here. We loop through all sources to get the tokenIds (versus just the pancake contract)